test/asm/charmap-inheritance.asm

```diff
@@ -0,0 +1,35 @@
+	; 'foo' should start with ASCII charmap entries
+	newcharmap foo
+
+	charmap "<START>",  $00
+	charmap "<RAM>",    $01
+	charmap "<WAIT>",   $02
+	charmap "<ASM>",    $03
+	charmap "<NUM>",    $04
+	charmap "<EXIT>",   $05
+	charmap "<SOUND>",  $06
+	charmap "<DAY>",    $07
+	charmap "<FAR>",    $08
+
+	; 'bar' should start with all the same charmap entries as foo,
+	; and 'newcharmap' should not segfault
```

Comment:
Wait, was the segfault in `newcharmap`? AIUI, the segfault occurred on the first function that referenced a charmap after `realloc`ing it, which is performed when adding a new mapping... did I misunderstand?

Reply:
Yes, the segfault was in `newcharmap`. `charmap_New` called `resizeCharmap`, but it had an old invalid value for `base`, so passing `base->capacity` to `charmap_New` gave a nonsensical value that caused `realloc` to fail. Keeping `base = charmap_Get(basename)` up-to-date with `hash_ReplaceElement` fixed that.

Reply:
Well, that would just be a consequence of referencing the pre-`realloc` value for the first time after `realloc`, not `charmap_New` itself. `base` isn't modified in `charmap_New`: the pointer passed to `resizeCharmap` is `NULL` in order to create a new charmap, like with `realloc`.

https://github.com/gbdev/rgbds/blob/f9daf27511bfa49a100fa45648c5098098d696b5/src/asm/charmap.c#L99

I think the description should reflect that instead. I also believe that the location of the added hashmap modification call supports what I described above.

Reply:
That matches my understanding of the segfault. I think a comment discussing the particular function calls that led to a segfault would belong in the code, like at the `hash_ReplaceElement` call, not in the test case where the only thing going on is `newcharmap`.

Reply:
My point is that referencing the `base` charmap in any other way (switching to it, then either adding a new mapping, or trying to translate a string) should also trigger a crash fixed by this PR, and that the comment should reflect that `newcharmap` isn't the entry point, as this could mislead a programmer into believing that `charmap_New` is at fault.

Reply:
Gotcha. I've updated the test case to reflect that.